common/remote/sockfile.hpp

```diff
@@ -50,11 +50,28 @@ enum ThrottleClass
 #define DEFAULT_SLOWCMD_THROTTLECPULIMIT 75
 #define DEFAULT_SLOWCMD_THROTTLEQUEUELIMIT 1000
 
-#define DEFAULT_AUTHORIZED_ONLY false
+
+enum RowServiceCfg
+{
+    rs_off,     // No dedicated row service, allows row service commands on std. dafilesrv port.
+    rs_on,      // Dedicated row service on own port accepting authorized signed connections only. Row service commands on std. dafilersv port will be refused.
+    rs_both,    // Dedicated row service on own port accepting authorized signed connections only. Still accepts unsigned connection on std. dafilesrv port
+    rs_onssl,   // Same as rs_on, but SSL
+    rs_bothssl  // Same as rs_only, but SSL
+};
```

Comment:
Also in comment above, rs_only -> rs_on ?

Reply:
agreed, spark-hpcc for example will be rowservice, but not always SSL based

Reply:
It had been set that it might only ever be SSL - since this is supposed to be a secure streaming service..

I think the main reason dafilesrv went down that route, is that SSL was added as an afterthought and there is no discovery mechanism.
So clients don't know how a target is configured, so e.g. Prod hitting Dev tries SSL, fails and tries non-SSL.

For this new service, somehow they need to know what to connect to, it could be hard-wired - I think the Spark side is at the moment (?)
OR they discover the info. they need to know, i.e. the port used for the service and whether secure connection or not.
NB: the fileservice could return this [minimal] info.

If a cluster is configured securely (and file access requests say so), then 3rd parties should only try to connect with SSL.
If request says it is not secure, then they can connect non-SSL.

Reply:
Can't think of compelling reason new service should it listen to both,
but there should be some mechanism for client to know what to connect to, which is true anyway.

Reply:
SSL connections won't always be possible, so clients coded to use rowservice are out of luck right?
Unless rowservice styled requests will be honored by the non-ssl port

Reply:
>Unless rowservice styled requests will be honored by the non-ssl port

Yes, if dafilesrv is configured like that, then the std. legacy port will allow insecure requests.

But I'm not sure I see any point if the RowService can accept insecure connections, why it would also listen to secure connections?
As long as the FileAccess service says it's non-ssl - and the client abides by what it says then the client will use it.
If FileAccess is says it's SSL, then it is secure and it won't be listening to an insecure port as well.

@rpastrana - Assuming FileAccess tells you whether secure(SSL) or not, can you provide an example (steps) where a Spark client might have issues?

Reply:
ok to be clear, if a client submits dafileserv requests using file access blog over non-ssl connection on the pre-existing port, those requests will be honored?
If that's the case, there's not much else to discuss.

on the other hand, if we set the policy that rowservice requests MUST be on ssl connections, there's nothing else to discuss either. However, there are situations where certs aren't available (not sure if that's enough to change your mind).

Reply:
back to @jakesmith